### PR TITLE
fix: calculate fiat amount from correct value

### PR DESCRIPTION
**Pull request by bumi**

### CODE REVIEW

File: src/app/screens/ConfirmPayment/index.tsx [hunk 1]
```diff
@@ -42,12 +42,26 @@ function ConfirmPayment(props: Props) {
...
       (async () => {
-        const res = await getFiatValue(budget);
-        setFiatAmount(res);
+        if (invoice.satoshis) {
+          const res = await getFiatValue(invoice.satoshis);
+          setFiatAmount(res);
+        }
+      })();
+    }
+  }, [invoice.satoshis, showFiat]);
```
Comment (lisabaut) [inline src/app/screens/ConfirmPayment/index.tsx:56]: Combine the dependency check with

```suggestion
     if (showFiat && invoice.satoshis) {
      (async () => {
          const res = await getFiatValue(invoice.satoshis);
          setFiatAmount(res);
      })();
    }
  }, [invoice.satoshis, showFiat]);
```

File: src/app/screens/ConfirmPayment/index.tsx [hunk 2]
```diff
@@ -42,12 +42,26 @@ function ConfirmPayment(props: Props) {
...
+
+  useEffect(() => {
+    if (showFiat) {
+      (async () => {
+        if (budget) {
+          const res = await getFiatValue(budget);
+          setFiatBudgetAmount(res);
+        }
```
Comment (lisabaut) [inline src/app/screens/ConfirmPayment/index.tsx:64]: Same here:

```suggestion
    if (showFiat && budget) {
      (async () => {
          const res = await getFiatValue(budget);
          setFiatBudgetAmount(res);
```

**Review by lisabaut — APPROVED** [review]
Thanks, See some suggestions below.

**Question:**
I am not sure what exactly this is fixing
but does this have something to do with this PR: [feat: show fiat in Confirm Payment](https://github.com/getAlby/lightning-browser-extension/pull/1324/files)
?

If so, is this also relevant then for ConfirmKeySend?

File: src/app/screens/ConfirmPayment/index.tsx (hunk 1 above)
Comment (bumi) [reply src/app/screens/ConfirmPayment/index.tsx:56]: lol, true

**Review by escapedcat — APPROVED** [review]

**Comment by bumi** [comment]
@lisabaut it has something to do with that. 
The wrong fiatValue has been used there. `fiatValue` is the fiat value of the suggested budget and NOT the fiat value of the payment amount.  The suggested budget is 10 times the invoice amount.

File: src/app/screens/ConfirmPayment/index.test.tsx [hunk 3]
```diff
@@ -79,12 +83,16 @@ describe("ConfirmPayment", () => {
       );
     });
 
+    expect(screen.getByText("25 sats")).toBeInTheDocument();
```
Comment (bumi) [inline src/app/screens/ConfirmPayment/index.test.tsx:86]: is this specific enough? because we have two fiat values in the document.

Comment (escapedcat) [reply src/app/screens/ConfirmPayment/index.test.tsx:86]: Made sure the second value is really not there in the beginning.

**Pull request merged**